modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/TreeFilter/ClayTreeFilter/ClayTreeFilter.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {TreeView as ClayTreeView} from '@clayui/core';
+import {ClayCheckbox} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import {cancelDebounce, debounce} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+
+import {
+	filterNodes,
+	getSelectedNodeObject,
+	handleNodeName,
+	selectedDataOutputTransfomer,
+	visit,
+} from './clayTreeUtils';
+
+const SEARCH_QUERY_MIN_LENGHT = 2;
+const SEARCH_INPUT_DEBOUNCE = 300;
+
+const TreeFilter = ({
+	childrenPropertyKey,
+	itemSelectorSaveEvent,
+	mandatoryFieldsForFiltering,
+	namePropertyKey,
+	nodes,
+	portletNamespace,
+}) => {
+	const [treeItems, setTreeItems] = useState(nodes);
+	const initialItemsRef = useRef(treeItems);
+
+	const [filterQuery, setFilterQuery] = useState('');
+	const [selectedItemsCount, setSelectedItemsCount] = useState(0);
+
+	const expanded = treeItems.length ? [treeItems[0].id] : [];
+
+	const [expandedKeys, setExpandedKeys] = useState(new Set(expanded));
+
+	const selectedNodesRef = useRef(null);
+	const refItemsCount = selectedNodesRef.current?.length || 0;
```

Comment:
It's just a small detail but I would add the word "selected" to this variable name so that we know that is the number of selected nodes.